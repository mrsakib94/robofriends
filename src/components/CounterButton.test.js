import CounterButton from './CounterButton';
import React from 'react';
import { shallow } from 'enzyme';

const mockColor = 'red';

describe('CounterButton', () => {
  it('renders without crashing', () => {
    expect(shallow(<CounterButton color={mockColor}/>)).toMatchSnapshot();
  });

  it('correnctly increments the counter', () => {
    const wrapper = shallow(<CounterButton color={mockColor}/>)
    expect(wrapper).toMatchSnapshot();
    wrapper.find('[id="counter"]').simulate('click');
    expect((wrapper.state())).toEqual({ count: 1 })
    wrapper.find('[id="counter"]').simulate('click');
    wrapper.find('[id="counter"]').simulate('click');
    expect((wrapper.state())).toEqual({ count: 3 });
    expect((wrapper.props().color)).toEqual(mockColor);
  });
});