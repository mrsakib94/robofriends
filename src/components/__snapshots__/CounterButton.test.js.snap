// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton correnctly increments the counter 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;

exports[`CounterButton renders without crashing 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
